modules/azure-public-dns/dns-records.tf

```diff
@@ -0,0 +1,164 @@
+data "azurerm_dns_zone" "dns_zone_entity" {
```

Comment:
Why is this data source here?
Either there should be a data source or a resource block of the same type. Let's discuss this today